src/dotnet/commands/dotnet-add-p2p/Program.cs

```diff
@@ -0,0 +1,232 @@
+        }
+
+        public static Func<ProjectItemElement, bool> IncludePred(string include)
+        {
+            return (it) => it.Include == include;
+        }
+
+        public static ProjectItemElement[] FindExistingItemsWithCondition(ProjectRootElement root, string framework, string include)
+        {
+            return root.Items
+                       .Where(
+                            AndPred(
+                                FrameworkPred<ProjectItemElement>(framework),
+                                IncludePred(include)))
+                       .ToArray();
+        }
+
+        public static ProjectItemGroupElement FindExistingUniformItemGroupWithCondition(ProjectRootElement root, string projectItemElementType, string framework)
+        {
+            return root.ItemGroupsReversed
+                            .FirstOrDefault(
+                                AndPred(
+                                    // When adding more predicates which operate on ItemGroup.Condition
+                                    // some slightly more advanced logic need to be used:
+                                    // i.e. ConditionPred(FrameworkConditionPred(framework), RuntimeConditionPred(runtime))
+                                    //   FrameworkConditionPred and RuntimeConditionPred would need to operate on a single condition
+                                    //   and ConditionPred would need to check if whole Condition Chain is satisfied
+                                    FrameworkPred<ProjectItemGroupElement>(framework),
+                                    UniformItemElementTypePred(projectItemElementType)));
+        }
+
+        public static ProjectItemGroupElement FindUniformOrCreateItemGroupWithCondition(ProjectRootElement root, string projectItemElementType, string framework)
+        {
+            var lastMatchingItemGroup = FindExistingUniformItemGroupWithCondition(root, projectItemElementType, framework);
+
+            if (lastMatchingItemGroup != null)
+            {
+                return lastMatchingItemGroup;
+            }
+
+            ProjectItemGroupElement ret = root.CreateItemGroupElement();
+            string condStr = GetFrameworkConditionString(framework);
+            if (condStr != null)
+            {
+                ret.Condition = condStr;
+            }
+
+            root.AppendChild(ret);
+            return ret;
+        }
+
+        public static void AddProjectToProjectReference(ProjectRootElement root, string framework, IEnumerable<string> refs)
+        {
+            const string ProjectItemElementType = "ProjectReference";
+
+            ProjectItemGroupElement ig = null;
+            foreach (var @ref in refs)
+            {
+                if (FindExistingItemsWithCondition(root, framework, @ref).Length == 0)
+                {
+                    Reporter.Output.WriteLine($"Item {ProjectItemElementType} including `{@ref}` is already present.");
```

Comment:
This is bleeding implementation a bit. Can we just say "Project already has a reference to {@ref}"?